datafusion/physical-plan/src/joins/sort_merge_join.rs

```diff
@@ -2032,7 +2034,10 @@ impl SortMergeJoinStream {
         let record_batch =
             concat_batches(&self.schema, &self.staging_output_record_batches.batches)?;
         self.join_metrics.output_batches.add(1);
-        self.join_metrics.output_rows.add(record_batch.num_rows());
+        let _ = self
+            .join_metrics
+            .baseline_metrics
+            .record_poll(Poll::Ready(Some(Ok(record_batch.clone()))));
```

Comment:
I would think cloning and returning an entire `record_batch` would be expensive, than just a single i64 that is `num_rows`

Reply:
Can we just `record_output(num_rows)` ?